src/openms/include/OpenMS/CHEMISTRY/NASequence.h

```diff
@@ -0,0 +1,504 @@
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Samuel Wein $
+// $Authors: Samuel Wein, Timo Sachsenberg $
+// --------------------------------------------------------------------------
+
+#ifndef OPENMS_CHEMISTRY_NASEQUENCE_H
+#define OPENMS_CHEMISTRY_NASEQUENCE_H
+
+#include <OpenMS/CHEMISTRY/EmpiricalFormula.h>
+#include <OpenMS/DATASTRUCTURES/String.h>
+#include <OpenMS/DATASTRUCTURES/Map.h>
+#include <OpenMS/CONCEPT/Types.h>
+#include <OpenMS/CHEMISTRY/Ribonucleotide.h>
+
+#include <vector>
+#include <iosfwd>
+
+namespace OpenMS
+{
+
+  class OPENMS_DLLAPI NASequence
+  {
+
+  public:
+    enum NASFragmentType
+    {                 //< NB: Not all fragments types are valid for all residue types, this class should probably get split
+      Full = 0,       //< with N-terminus and C-terminus
+      Internal,       //< internal, without any termini
+      FivePrime,      //< only 5' terminus
+      ThreePrime,     //< only 3' terminus
+      AIon,           //< MS:1001229 N-terminus up to the C-alpha/carbonyl carbon bond
+      BIon,           //< MS:1001224 N-terminus up to the peptide bond
+      CIon,           //< MS:1001231 N-terminus up to the amide/C-alpha bond
+      XIon,           //< MS:1001228 amide/C-alpha bond up to the C-terminus
+      YIon,           //< MS:1001220 peptide bond up to the C-terminus
+      ZIon,           //< MS:1001230 C-alpha/carbonyl carbon bond
+      Precursor,      //< MS:1001523 Precursor ion
+      BIonMinusH20,   //< MS:1001222 b ion without water
+      YIonMinusH20,   //< MS:1001223 y ion without water
+      BIonMinusNH3,   //< MS:1001232 b ion without ammonia
+      YIonMinusNH3,   //< MS:1001233 y ion without ammonia
+      NonIdentified,  //< MS:1001240 Non-identified ion
+      Unannotated,    //< no stored annotation
+      WIon,           //< W ion, added for nucleic acid support
+      AminusB,        //< A ion with base loss, added for nucleic acid support
+      DIon,           //< D ion, added for nucleic acid support
```

Comment:
all of these need to be `///<` (note the three slashes) for doxygen to pick them up.